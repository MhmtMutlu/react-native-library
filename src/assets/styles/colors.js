const colors = {
  grey: '#6c757d',
  darkGrey: '#495057',
  darkGrey2x: '#343a40',
  lightGrey: '#adb5bd',
  lightGrey2x: '#dee2e6',
  lightGrey3x: '#e0e0e0',
  white: '#f8f9fa',
};

export default colors;
