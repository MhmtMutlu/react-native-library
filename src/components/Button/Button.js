import React from 'react';
import {TouchableOpacity, Text, ActivityIndicator} from 'react-native';

import styles from './Button.styles';

const Button = ({text, onPress, loading, theme = 'primary'}) => {
  return (
    <TouchableOpacity
      testID="button-touchable"
      style={styles[theme].container}
      onPress={onPress}
      disabled={loading}>
      {loading ? (
        <ActivityIndicator testID="button-indicator" color="white" />
      ) : (
        <Text testID="button-text" style={styles[theme].title}>
          {text}
        </Text>
      )}
    </TouchableOpacity>
  );
};

export default Button;
