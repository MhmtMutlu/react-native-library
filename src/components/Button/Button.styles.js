import {StyleSheet} from 'react-native';
import colors from '../../assets/styles/colors';

const base_style = StyleSheet.create({
  container: {
    padding: 10,
    margin: 5,
    alignItems: 'center',
    borderRadius: 10,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 17,
  },
});

export default {
  primary: StyleSheet.create({
    ...base_style,
    container: {
      ...base_style.container,
      backgroundColor: colors.darkGrey2x,
    },
    title: {
      ...base_style.title,
      color: colors.white,
    },
  }),
  secondary: StyleSheet.create({
    ...base_style,
    container: {
      ...base_style.container,
      backgroundColor: colors.lightGrey3x,
    },
    title: {
      ...base_style.title,
      color: colors.darkGrey2x,
    },
  }),
};
