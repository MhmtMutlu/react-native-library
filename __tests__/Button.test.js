import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from '../src/components/Button';
import styles from '../src/components/Button/Button.styles';

test('should match with snapshot', () => {
  const comp = render(<Button />);
  expect(comp).toMatchSnapshot();
});

test('should render text correctly', () => {
  const testText = 'test';
  const comp = render(<Button text={testText} />);
  const buttonText = comp.getByTestId('button-text').children[0];
  expect(buttonText).toBe(testText);
});

test('should trigger onPress', () => {
  const mockFunc = jest.fn();
  const comp = render(<Button onPress={mockFunc} />);
  const buttonTouchable = comp.getByTestId('button-touchable').children[0];
  fireEvent(buttonTouchable, 'press');
  expect(mockFunc).toBeCalled();
});

test('should render given theme', () => {
  const testTheme = 'primary';
  const comp = render(<Button theme={testTheme} />);
  const buttonStyle = comp.getByTestId('button-touchable').props.style;
  expect(buttonStyle).toMatchSnapshot(styles[testTheme].container);
});

test('should render loading', () => {
  const testLoading = true;
  const comp = render(<Button loading={testLoading} />);
  const buttonIndicator = comp.getByTestId('button-indicator');
  expect(buttonIndicator).toBe(buttonIndicator);
});
