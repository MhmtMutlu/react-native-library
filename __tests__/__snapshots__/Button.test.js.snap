// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#343a40",
      "borderRadius": 10,
      "margin": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Object {
        "color": "#f8f9fa",
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
    testID="button-text"
  />
</View>
`;

exports[`should render given theme 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#343a40",
  "borderRadius": 10,
  "margin": 5,
  "opacity": 1,
  "padding": 10,
}
`;
